### DISCUSSION TITLE
Any way of triggering a PC-based call prompted by an action on a hub?

### DISCUSSION THREAD

**Discussion by BirchBirch** [pybricks/support#1950] [Ideas]
Hi,

I'd love to be able to trigger the playing of audio files on the hosting PC according to actions on a hub.

I'm well aware that the programs run locally on the hub though. I'm using the City and Move hubs and there is no audio capability on those of course, but is there **scope** to extend the front end so that print commands (containing embedded codes perhaps) could be interpreted and run on the PC when it's connected?

We're deep into the family's annual Christmas project of building an extended winter village, which includes multiple hub-driven trains and trams. In previous years we've had to switch backwards and forwards on Lego's app, but now we want to consolidate control into one place *and* add in things like point switching and synchronised crossing gates. The gkids love the sounds on the Lego apps, hence my request so that we can include these in a Pybricks-controlled display as well.

[My thinking is that although the app is running on the hub, the front end is still connected, displaying the printed output and allowing the 'stop program' command. If something could interpret the incoming print messages, looking for a key string (say, "##SysCall <parameter>") that could then invoke a PC-side call...]

Just thinking aloud...
Cheers,
Kev

**Comment by BertLindeman** [comment]
Hello,

Did you take a look at [pybricks hub-to-device pc-communication/](https://pybricks.com/projects/tutorials/wireless/hub-to-device/pc-communication/) to get an idea?

Bert

**Reply by BirchBirch** [reply]
Thanks, Bert!

I hadn't found the hub-to-device comms page - too far down the tutorials nav!

That seems to be the way to go! I'm now receiving prompts from the hub back on the PC so now I can start to trigger the playing of some sounds!

(BTW, I'll write up the coding once I've got it working)

Cheers,
Kev
